sites/public/pages/applications/start/autofill.tsx

```diff
@@ -0,0 +1,126 @@
+import { useForm } from "react-hook-form"
+import FormsLayout from "../../../layouts/forms"
+import { useFormConductor } from "../../../lib/hooks"
+import FormSummaryDetails from "../../../src/forms/applications/FormSummaryDetails"
+import AutofillCleaner from "../../../lib/appAutofill"
+
+export default () => {
+  const context = useFormConductor("autofill")
+  const { conductor, application, listing } = context
+  const { initialStateLoaded, profile } = useContext(UserContext)
+  const { applicationsService } = useContext(ApiClientContext)
+  const [submitted, setSubmitted] = useState(false)
+  const [previousApplication, setPreviousApplication] = useState<Application>(null)
+
+  const currentPageSection = 1
+  let useDetails = false
+
+  /* Form Handler */
+  const { handleSubmit } = useForm()
+  const onSubmit = () => {
+    if (!submitted) {
+      // Necessary to avoid infinite rerenders
+      setSubmitted(true)
+      if (previousApplication && useDetails) {
+        conductor.application = previousApplication
+      } else {
+        conductor.application = blankApplication()
+      }
+      context.syncApplication(conductor.application)
+      conductor.sync()
+      conductor.routeToNextOrReturnUrl()
+    }
+  }
+
+  const fetcher = async () => {
+    const res = await applicationsService.list({
+      userId: profile.id,
+      orderBy: "createdAt",
+      order: "DESC",
+      limit: 1,
+    })
+    return res
+  }
+  const { data } = useSWR(
+    profile && !previousApplication ? process.env.listingServiceUrl : null,
+    fetcher
+  )
+  if (data) {
+    if (data.items.length > 0) {
+      setPreviousApplication(new AutofillCleaner(data.items[0]).clean())
+    } else {
+      onSubmit()
+    }
+  } else if (initialStateLoaded && !profile) {
+    onSubmit()
+  }
+
+  return previousApplication ? (
+    <FormsLayout>
+      <FormCard header={listing?.name}>
+        <ProgressNav
+          currentPageSection={currentPageSection}
+          completedSections={application.completedSections}
+          labels={conductor.config.sections.map((label) => t(`t.${label}`))}
+        />
+      </FormCard>
+
+      <FormCard>
+        <div className="form-card__lead border-b">
+          <h2 className="form-card__title is-borderless mt-4">
+            {t("application.autofill.saveTime")}
+          </h2>
+        </div>
+        <div className="form-card__pager-row px-16">
+          <p className="field-note py-2">{t("application.autofill.prefillYourApplication")}</p>
+        </div>
+        <FormSummaryDetails
+          application={previousApplication}
+          editMode={false}
+          hidePreferences={true}
+        />
+        <Form onSubmit={handleSubmit(onSubmit)}>
+          <div className="form-card__pager">
+            <div className="form-card__pager-row primary">
+              <Button
+                styleType={AppearanceStyleType.primary}
+                onClick={() => {
+                  useDetails = true
```

Comment:
Shouldn't be processed by React and `useState`?

Reply:
Well it's not a state change, it's just a runtime flag that lets the submit handler know which action to take. I specifically wouldn't want to trigger any re-render.